instrumentation/que/lib/opentelemetry/instrumentation/que/middlewares/server_middleware.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Que
+      module Middlewares
+        # Server middleware to trace Que jobs
+        class ServerMiddleware
+          def self.call(job, &block)
+            job_class = job.que_attrs[:job_class] || job.class.name
+
+            tracer.in_span(
+              "#{job_class} process",
+              attributes: attributes_before_job_complation(job, job_class),
+              kind: :consumer
+            ) do |span|
+              block.call
+
+              span.set_attribute('messaging.que.attempts', job.que_attrs[:error_count])
+              if (error = job.que_error)
+                span.record_exception(error)
+                span.status = OpenTelemetry::Trace::Status.error("Unhandled exception of type: #{error.class}")
```

Comment:
The general guidelines suggest that instrumentation should not set `span.status` to `error` - only application code should do that. That said, I *think* this is OK, since we're pretty sure at this point that the job's error is "real".